### PR TITLE
QA example refactoring.

### PR DESCRIPTION
**Pull request by devrimcavusoglu**
- Notebook updated as to match new version.

### CODE REVIEW

File: trapper/data/data_processors/squad/question_answering_processor.py [hunk 1]
```diff
@@ -47,15 +47,14 @@ def filtered_instance() -> IndexedInstance:
...
             "context": [-1],
-            "qa_id": -1,
+            "qa_id": "",
             "question": [-1],
             "__discard_sample": True,
         }
 
     def text_to_instance(
-        self, context: str, question: SpanTuple, id_: str, answer: SpanTuple = None
+        self, context: str, question: SpanTuple, id: str, answer: SpanTuple = None
```
Comment (cemilcengiz) [inline trapper/data/data_processors/squad/question_answering_processor.py:56]: id is a library function, I think overwriting it this way is not nice.

File: trapper/data/data_processors/squad/question_answering_processor.py [hunk 2]
```diff
@@ -72,7 +71,7 @@ def text_to_instance(
             )
             instance.update(indexed_answer)
 
-        instance["qa_id"] = id_
+        instance["qa_id"] = id
```
Comment (cemilcengiz) [inline trapper/data/data_processors/squad/question_answering_processor.py:74]: (again) id is a library function, I think overwriting it this way is not nice.

File: trapper/pipelines/question_answering_pipeline.py [hunk 3]
```diff
@@ -68,17 +68,21 @@ def normalize(self, item):
...
 
     @staticmethod
     def _convert_to_span_tuple(span: Union[SpanDict, SpanTuple]) -> SpanTuple:
-        if isinstance(span, SpanDict):
+        if isinstance(span, dict):
             span = convert_spandict_to_spantuple(span)
         return span
 
+    @staticmethod
+    def _add_id(item: Dict) -> None:
```
Comment (cemilcengiz) [inline trapper/pipelines/question_answering_pipeline.py:82]: why do we do that? "id" is fine as a key.

Comment (devrimcavusoglu) [reply trapper/pipelines/question_answering_pipeline.py:82]: we pass as `(**example)`, I did not want to change that, so i manipulate `id` as `id_` for argument name `id_`.

**Review by cemilcengiz — APPROVED** [review]
LGTM

**Pull request merged**